server/rules-engine/index.js

```diff
@@ -0,0 +1,172 @@
+const DEBUG = false
+const Engine = require('json-rules-engine').Engine
+const enums = require('./enums')
+const Rule = require('json-rules-engine').Rule
+const operators = require('./operators')
+const factHandler = require('./factHandler')
+
+const _buildStandardRules = function (conditions, ruleName, eventType) {
+  if (DEBUG) console.log('Setting up standard rules with conditions ', conditions)
+  return conditions.map(condition => new Rule({
+    conditions: {
+      all: [condition]
+    },
+    event: {
+      type: eventType
+    },
+    name: ruleName,
+    priority: 5
+  }))
+}
+
+const _setupStandardRule = function (conditions, ruleName, eventType) {
+  const rules = _buildStandardRules(conditions, ruleName, eventType)
+  for (const rule of rules) {
+    this.engine.addRule(rule)
+  }
+  this.engine.on('success', function (event, almanac, ruleResult) {
+    if (event.type === eventType) {
+      almanac.addRuntimeFact(enums.ruleRejected, true)
+      if (DEBUG) {
+        almanac.factValue('ref').then(ref => {
+          console.log('Rule passed, rejecting parcel:', ref)
+        })
+      }
+    }
+  })
+}
+
+const _buildAcceptedItemsRules = function () {
+  return [new Rule({
+    conditions: {
+      all: [{
+        fact: enums.ruleRejected,
+        operator: 'notEqual',
+        value: true
+      }]
+    },
+    event: {
+      type: enums.acceptedEventName
+    },
+    priority: 1,
+    name: 'matchAcceptedItems'
+  })]
+}
+
+const _setupAcceptedItemsRule = function (funcToCall) {
+  const rules = _buildAcceptedItemsRules()
+  for (const rule of rules) {
```

Comment:
Adding a collection of rules is a common pattern, maybe add an `engine.addRules` method?

Reply:
Added a method to do this, and was able to simplify a `for ()` loop into a `map` as well.